### ISSUE TITLE
Linking to Reactome Pathway?

### ISSUE THREAD

**Issue by sibyl229** [WormBase/website#7293]
Hi @chris-grove and @WBjae 

Someone from Reactome (https://reactome.org/), a curated pathway database gave a talk at the lab meeting today. They automatically infer C elegans pathways from curated human pathway based on orthology. 

I'm wondering if we would have an interest in linking to their resource. Here is an example pathway: https://reactome.org/PathwayBrowser/#/R-CEL-373760

They also provide mappings from UniProt identifiers and a few other identifiers to the Reactome pathway identifiers, which potentially allows us to link to their resource from the WormBase website. Here are the mappings: https://reactome.org/download-data

What do you think?

**Comment by chris-grove** [comment]
@sibyl229 
Cool! Sure, we could link to these pathways via the External Links widget. It would be nice if there was a way to view the pathways with different labels/names for the gene products, as currently the gene products/proteins are labeled with a human protein name like "SH3GL2" rather than the C. elegans protein name like "UNC-57", but that's probably a project in itself.

**Comment by tharris** [comment]
We had links to reactome (and it’s predecessor) for a long time, but I believe these links were built dynamically and never formally stored under Database. If we have the uniprot mappings it should be easy to generate associations between WB and Reactome.

**Comment by paulwsternberg** [comment]
Links are a good start, and will highlight what reactome needs to do....

Paul

On Sep 11, 2019, at 11:52 AM, Chris Grove <notifications@github.com<mailto:notifications@github.com>> wrote:


@sibyl229<https://github.com/sibyl229>
Cool! Sure, we could link to these pathways via the External Links widget. It would be nice if there was a way to view the pathways with different labels/names for the gene products, as currently the gene products/proteins are labeled with a human protein name like "SH3GL2" rather than the C. elegans protein name like "UNC-57", but that's probably a project in itself.

**Comment by khowe** [comment]
Reactome provide a widget that can be embedded into other sites. Ensembl make use of this:

http://plants.ensembl.org/Triticum_aestivum/Gene/Pathway?db=core;g=TraesCS1B02G195200;r=1B:350391339-350397150;t=TraesCS1B02G195200.1

But as commented, the gene product names are not WormBase-friendly.

**Comment by WBjae** [comment]
Hi Sibyl,
Regarding this, can we get some analysis for the interaction data with
Reactome pathway like below?
This is string-db.org page for wago-2 interaction network.

Thanks,
Jae

[image: image.png]

**Comment by sibyl229** [comment]
Wow! Thank you all for your thoughts on this! 

@chris-grove @paulwsternberg @khowe Looks like the human gene product name is a bit problematic here. Let me talk with Reactome developers at OICR to see if this can be addressed.

@tharris @azurebrd @khowe I think it would be nice if Reactome could resolve and redirect an Reactome URL based on UniProt ID, instead of the Reactome pathway ID. But I imagine it's a lot to ask of them. Is bringing into WB the Reactome pathway ID and attaching them to a gene production (based on a UniProt ID to Reactome pathway ID mapping that Reactome provides) something we have a process for at WB? (I'm not too sure who to address this question to 😄)

@khowe this is quite neat! I will ask the Reactome devs on how to have something similar setup, and whether any incompatibility would occur between WB site and the widget.

@WBjae the image you attached didn't seem to be included with your message. You might need to upload it here: https://github.com/WormBase/website/issues/7293. I can put you in touch with one of the Reactome devs here, if you are interested.

**Comment by sibyl229** [comment]
Hi, an update on linking to Reactome and some questions. 

I had some discussion with Joel, who is a seasoned developer at Reactome-OICR. While Joel is enthusiastic about linking WormBase and Reactome, **one challenge is that Reactome doesn't have worm protein names in their database**. So, as many of you pointed out, the Reactome inferred C. elegans pathway display the human protein names. This likely will cause confusion with worm researchers, reducing the usefulness of the pathways. Joel mentioned that there were discussions about expanding their inference code to include gene name, and he would bring it up again.

I asked Joel for an estimate on the effort required for this task, and I will keep you posted.

One question for this group, what would be your preferred source for protein names (or gene name, if that's preferred). Reactome uses the UniProt ID to identify a worm gene product, and it appears that the UniProt protein name can be different from the WormBase protein name.

Thanks!

**Comment by khowe** [comment]
UniProt entries often have lots of names associated with them. Example:

https://www.uniprot.org/uniprot/G5ECL2.txt

We have SYNJ_CAEEL (ID), G5ECL2 (accession), Synaptojanin (Full record name), Synaptic inositol 1,4,5-trisphosphate 5-phosphatase (alternative full name), Uncoordinated protein 26 (another alternative full name), unc-26 (gene name).

I guess the name that would be most familiar with WormBase users is the gene name (GN line). However, it would be really nice to be able to toggle between different types of name/id. 


ID   SYNJ_CAEEL              Reviewed;        1119 AA.
AC   G5ECL2; B3CJ53; G5EDR8; G5EEJ9;
DE   RecName: Full=Synaptojanin {ECO:0000303|PubMed:10931870};
DE            EC=3.1.3.36 {ECO:0000250|UniProtKB:Q9D2G5};
DE   AltName: Full=Synaptic inositol 1,4,5-trisphosphate 5-phosphatase {ECO:0000250|UniProtKB:Q9D2G5};
DE   AltName: Full=Uncoordinated protein 26 {ECO:0000312|WormBase:JC8.10b};
GN   Name=unc-26 {ECO:0000312|WormBase:JC8.10b};
GN   ORFNames=JC8.10 {ECO:0000312|WormBase:JC8.10b};

**Comment by sibyl229** [comment]
@khowe Thank you! I will let Joel know about the name preference and the ability to toggle between different names. 

There is one difference, probably a minor one, that UniProt doesn't say "isoform b" anywhere, unlike WormBase.  But UniProt has information on the CDS, "JC8.10b". Do you think the "JC8.10b" or "isoform b" should be parsed out for the default name display? Thanks!

**Comment by sibyl229** [comment]
Hi, more updates on linking to Reactome pathways. 

Reactome is working on getting C. elegans gene names from UniProt and incorporating those into their database and display. I was told that for protein complexes, getting the name fixed would be harder than individual proteins. So they plan on having individual proteins working first, and then work on protein complexes in a separate release later on. I will ask them for a more detailed timeline soon.

@khowe @chris-grove @WBjae @azurebrd If we are to incorporate Reactome pathway ID into WormBase, what do you need from me? 

I think the expected outcome, if you all (especially @chris-grove and @WBjae agree), is that 
-  in the short term, add a link to Reactome pathway page from the interactions widget on gene page
- in the long term, after we confirm their display name issue is fixed, embed a Reactome pathway graphic (like @khowe linked above) in interaction or another widget

Here is what I know about the Reactome:
- They suggest linking to their site using Reactome pathway IDs. They _don't currently_ support a UniProt ID based URL.
- Reactome provide a file containing a mapping of UniProt ID to Reactome pathway ID
- UniProt ID and Reactome pathway ID is a many-to-many relation (in particular, it's possible for a single protein to be in multiple pathways)
- Reactome also operates on a release cycle on months each

**Comment by chris-grove** [comment]
@sibyl229 
OK, so it seems we would first need to generate our own mappings of WBGene ID to UniProtID to Reactome Pathway? I suppose that's trivial and can just be handled in the code (as we should already have WB protein-to-UniProt mappings)?

Once we have that I think we could provide hyperlinks to relevant Reactome pathways from the Interactions widget as well as possibly the External Links widget. Maybe the links could be labeled with the pathway name?

Yes, a single gene could potentially link to many Reactome pathways. We would also need to make sure it is clear to our users that these are not curated worm pathways, but rather human pathways that have been extrapolated to worms via orthology to the respective human genes.

**Comment by sibyl229** [comment]
@chris-grove good catch that we also need to track the pathway names (which is available in the same mapping file provided by Reactome) in WB databases.

For the UI, we potentially can have something like (rough draft):

Orthology-based inferred pathway <link>mRNA splicing</link>        Powered by <link>Reactome Logo</link>

**Comment by chris-grove** [comment]
@sibyl229 
Yes, that sounds good. If possible it would be good to have some help text available explaining exactly what we mean by that.

**Comment by sibyl229** [comment]
@chris-grove Reactome documents their algorithm for inference here: https://reactome.org/documentation/inferred-events. We can probably summarize it and then link to it for those who need more details. I can make a draft later if needed.

The key question now is how to get the mapping to Reactome pathway ID into WormBase database. I imagine it needs a model amendment , the actual parsing of the file, and probably other things I don't know about. I'm wondering how to go about this, what can I do to make this easier for everybody involved.

**Comment by WBjae** [comment]
@sibyl229 
If possible, can you provide us a mock-up images showing using examples by users for the Reactome link?

**Comment by sibyl229** [comment]
Hi @WBjae that's a good point! Thanks!

Here is what I imagine it would look like in the short term:
![Screen Shot 2019-11-07 at 5 28 46 PM](https://user-images.githubusercontent.com/1753893/68433360-5022f300-0184-11ea-97c5-6b8b1f56f271.png)

The tooltip (when mouseover the field label) might say something like: "Computationally inferred pathways from curated pathways in humans based on protein homology"

**Comment by chris-grove** [comment]
@sibyl229 
OK, yes that looks good. One simple approach (although I don't know if it will work) would be to use the existing "Database" tag in the ?Gene model:

?Gene
DB_info   Database   ?Database   ?Database_field   ?Text

which for the sem-5 gene, for example, look like

AceView | gene | XI578
WormQTL | gene | WBGene00004774
SignaLink | protein | WBGene00004774
WormFlux | gene | WBGene00004774
NDB | locus_tag | CELE_C14F5.5

so we could have something like:

Reactome | <Reactome_field> | <Reactome_pathway_ID>

and then possibly generate the display based on that? That would also populate the External Links widget, I believe. Otherwise we might want a separate tag in the ?Gene model like

?Gene
Pathway   Reactome_pathway  ?Text

with the option to add pathway-database-specific tags for other pathway databases. As far as display is concerned I guess the Reactome links would go in the Interactions widget? Or maybe in a a new and separate Pathways widget or something? If we go with the latter we could include information about WormBase ?WBProcess info connected to the gene.

**Comment by sibyl229** [comment]
Thanks @chris-grove for working out the schema. I like the idea that the Reactome pathway IDs are included in the external link section without schema changes.

The tricky issue is that if we would like the include the Reactome pathway name (like "Vitamin B5 metabolism), it seems that the Database tag isn't able to handle this additional piece of information. I'm not sure whether this is true, if so, how to solve the problem.

Maybe we can put the Reactome links in the Interactions widget for now. Though I think it's interesting to think about WBProcess along with interactions.
